node-red-node-wot/src/wot-property.js

```diff
@@ -1,174 +1,176 @@
+                    node.status({
+                        fill: "red",
+                        shape: "ring",
+                        text: "Observe error",
+                    })
+                }
+                if (ob) {
+                    node.status({
+                        fill: "green",
+                        shape: "dot",
+                        text: "connected",
+                    })
+                    break
+                }
+                await (() => {
+                    return new Promise((resolve) => {
+                        setTimeout(() => {
+                            resolve()
+                        }, 500)
+                    })
+                })()
+            }
         })
-    })
 
-    node.on('close', function () {
-      if (removed) {
-        // This node has been deleted
-      } else {
-        // This node is being restarted
-      }
-      done()
-    })
-  }
-  RED.nodes.registerType('read-property', readPropertyNode)
+        node.on("input", async (msg, send, done) => {
+            if (!consumedThing) {
+                node.error("[error] consumedThing is not defined.")
+                done("consumedThing is not defined.")
+                return
+            }
+            const uriVariables = config.uriVariables ? JSON.parse(config.uriVariables) : undefined
+            consumedThing
+                .readProperty(config.property, { uriVariables: uriVariables })
+                .then(async (resp) => {
+                    let payload
+                    try {
+                        payload = await resp.value()
+                    } catch (err) {
+                        console.error(`[error] failed to get value. name: ${config.property}, error: `, err.toString())
+                        done(`[error] failed to get value. name: ${config.property}, error: ${err.toString()}`)
+                    }
+                    node.send({ payload, topic: config.topic })
+                    node.status({
+                        fill: "green",
+                        shape: "dot",
+                        text: "connected",
+                    })
+                    done()
+                })
+                .catch((err) => {
+                    node.warn(err)
+                    node.status({
+                        fill: "red",
+                        shape: "ring",
+                        text: "Response error",
+                    })
+                    done(err)
+                })
+        })
 
-  function writePropertyNode(config) {
-    RED.nodes.createNode(this, config)
-    let node = this
+        node.on("close", function () {
+            // eslint-disable-next-line
+            if (removed) {
+                // This node has been deleted
+            } else {
+                // This node is being restarted
+            }
+            // eslint-disable-next-line
+            done()
+        })
+    }
```

Comment:
@hidetak please let me know how I can remove the eslint ignore comments.

Reply:
@egekorkan 
The callback arguments `removed` and `done` were missing.
My apologies for missing that.

https://github.com/eclipse-thingweb/node-red/pull/4/files/ab1be6a993ae86e8f666114d5f687c7e40aa2092#diff-52388816cc91b77d02014cdacab7478e4777a58db9195c95c1b9ad3baf295e15R116

Reply:
Thanks!